File: 2347-best-poker-hand/2347-best-poker-hand.cpp
class Solution {
public:
    string bestHand(vector<int>& ranks, vector<char>& suits) 
    {
     map<int,int>mp;
        for(auto x:suits)
        {
            mp[x]++;
        }
        map<char,int>m;
        for(auto x:ranks)
        {
            m[x]++;
        }
        for(auto [u,v]:mp)
        {
            if(v==5)
            {
                return "Flush";
            }
            
        }
        for(auto [u,v]:m)
        {
            if(v>=3)
            {
                return "Three of a Kind";
            }
           
        }
          for(auto [u,v]:m)
        {
            if(v==2)
            {
                return "Pair";
            }
           
        }
        return "High Card";
    }
};

File: 2347-best-poker-hand/2347-best-poker-hand.ts
function bestHand(ranks: number[], suits: string[]): string 
{
       if(new Set(suits).size===1) return "Flush";
    
         let mp=new Map();
    
         let mx=0;
         for(let i=0;i<ranks.length;i++)
        {
            mp.set(ranks[i],(mp.get(ranks[i])||0)+1)
            mx= Math.max(mx,mp.get(ranks[i]));
        }
       if(mx>=3)
           {
               return "Three of a Kind";
           }
      if(mx==2)
          {
              return "Pair";
          }
      return "High Card"
};